analyzer_test.go

```diff
@@ -111,9 +117,33 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
 				} {
 					got := h.MustReadFile(t, filepath.Join(layerDir, data.name))
 					h.AssertStringContains(t, string(got), data.want)
+					h.AssertStringContains(t, string(got), unsetFlags)
 				}
 			})
 
+			when("buildpack API < 0.6", func() {
```

Comment:
I think we might have an edge case where the previous <layer>.toml was written by a buildpack with the same id but implementing a different api. In this case I think we should ensure that we restore the file in the format that the current buildpack is expecting to find it. This might involve some changes to ensure that we're using the right encoder & decoder as they could be different.

Reply:
I don't think that we have any problem here because we didn't change the `.json` format. So in your example, the `analyzer` will read the `.json` file of the old buildpack and will write the `.toml` file in the new format.